src/worklogs/WorkLogs.js

```diff
@@ -78,78 +78,54 @@ export default compose(
       openModal,
       clearSelection,
       setJiraIssueForSelection
-    }) => e => {
-      if (mode === MODES.LOGGING) {
-        if (e.key === 'Escape') {
-          switchMode(MODES.NORMAL)
+    }) => ({
+      keyBindings: {
+        [MODES.LOGGING]: {
+          plainKey: { Escape: () => switchMode(MODES.NORMAL) }
+        },
+        [MODES.NORMAL]: {
+          plainKey: {
+            l: () => switchMode(MODES.LOGGING),
+            d: () => openModal(MODAL_TYPES.EDIT_DESCRIPTION),
+            t: () => openModal(MODAL_TYPES.EDIT_TIME),
+            i: () => openModal(MODAL_TYPES.INSERT_POINT_IN_TIME),
+            s: () => openModal(MODAL_TYPES.SET_JIRA_ISSUE),
+            n: () => setJiraIssueForSelection(null),
+            ArrowUp: () => moveCursor(-1),
+            ArrowDown: () => moveCursor(1),
+            k: () => moveCursor(-1),
+            j: () => moveCursor(1),
+            Backspace: deleteSelection,
+            Delete: deleteSelection,
+            ' ': toggleSelectCursorPoint
+          },
+          withCtrlKey: {
+            '-': clearSelection
+          },
+          withShiftKey: {
+            ArrowUp: () => moveSelection(-1),
+            ArrowDown: () => moveSelection(1),
+            k: () => moveSelection(-1),
+            j: () => moveSelection(1)
+          }
         }
-        return
       }
-
-      // normal mode
-      switch (e.key) {
-        case 'l':
-          switchMode(MODES.LOGGING)
-          break
-
-        case 'd':
-          openModal(MODAL_TYPES.EDIT_DESCRIPTION)
-          break
-
-        case 'e':
-          openModal(MODAL_TYPES.EDIT_TIME)
-          break
-
-        case 'i':
-          openModal(MODAL_TYPES.INSERT_POINT_IN_TIME)
-          break
-
-        case 't':
-          openModal(MODAL_TYPES.SET_JIRA_ISSUE)
-          break
-
-        case 'n':
-          setJiraIssueForSelection(null)
-          break
-
-        case 'Backspace':
-        case 'Delete':
-          deleteSelection()
-          break
-
-        case ' ':
-          toggleSelectCursorPoint()
-          break
-
-        case '-':
-          if (!e.ctrlKey) {
-            break
-          }
-          clearSelection()
-          break
-
-        case 'ArrowUp':
-          if (e.shiftKey) {
-            moveSelection(-1)
-          } else {
-            moveCursor(-1)
-          }
-          break
-
-        case 'ArrowDown':
-          if (e.shiftKey) {
-            moveSelection(1)
-          } else {
-            moveCursor(1)
-          }
-          break
-
-        default:
-          break
+    })
+  ),
+  withHandlers({
+    handleKeyDown: ({ mode, keyBindings }) => e => {
+      const metaKeyPath =
```

Comment:
I would put this into a simple helper function.
```suggestion
      const metaKeyPath = getMetaKeyPath(e)
```